presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSource.java

```
@@ -0,0 +1,242 @@
+    {
+        if (valueString == null) {
+            return null;
+        }
+
+        try {
+            if (type.equals(BOOLEAN)) {
+                if (valueString.equalsIgnoreCase("true")) {
+                    return true;
+                }
+                if (valueString.equalsIgnoreCase("false")) {
+                    return false;
+                }
+                throw new IllegalArgumentException();
+            }
+            if (type.equals(INTEGER)) {
+                return parseLong(valueString);
+            }
+            if (type.equals(BIGINT)) {
+                return parseLong(valueString);
+            }
+            if (type.equals(REAL)) {
+                return (long) floatToRawIntBits(parseFloat(valueString));
+            }
+            if (type.equals(DOUBLE)) {
+                return parseDouble(valueString);
+            }
+            if (type.equals(DATE)) {
+                return parseLong(valueString);
+            }
+            if (type instanceof VarcharType) {
+                Slice value = utf8Slice(valueString);
+                return value;
+            }
+            if (type.equals(VarbinaryType.VARBINARY)) {
+                return utf8Slice(valueString);
+            }
+            if (isShortDecimal(type) || isLongDecimal(type)) {
+                DecimalType decimalType = (DecimalType) type;
+                BigDecimal decimal = new BigDecimal(valueString);
+                decimal = decimal.setScale(decimalType.getScale(), BigDecimal.ROUND_UNNECESSARY);
+                if (decimal.precision() > decimalType.getPrecision()) {
+                    throw new IllegalArgumentException();
+                }
+                BigInteger unscaledValue = decimal.unscaledValue();
+                return isShortDecimal(type) ? unscaledValue.longValue() : Decimals.encodeUnscaledValue(unscaledValue);
+            }
+        }
+        catch (IllegalArgumentException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)